ivy/array/array.py

```diff
@@ -135,10 +135,47 @@ def _init(self, data):
         else:
             self._post_repr = ")"
         self.backend = ivy.current_backend_str()
+        if dynamic_backend is not None:
+            self._dynamic_backend = dynamic_backend
+        else:
+            self._dynamic_backend = ivy.get_dynamic_backend()
 
     # Properties #
     # ---------- #
 
+    @property
+    def dynamic_backend(self):
+        return self._dynamic_backend
+
+    @dynamic_backend.setter
+    def dynamic_backend(self, value):
+        from ivy.functional.ivy.gradients import _variable
+        from ivy.backend_handler import _determine_backend_from_args
+
+        if value == False:
+            self._backend = _determine_backend_from_args(self)
+
+        else:
+            is_variable = self._backend.is_variable
+            to_numpy = self._backend.to_numpy
+            variable_data = self._backend.variable_data
+
+            if is_variable(self.data) and \
+                    not ( str(self._backend).__contains__("jax") or
+                          str(self._backend).__contains__("numpy")
+                    ):
+                native_data = variable_data(self.data)
+                np_data = to_numpy(native_data)
+                new_arr = ivy.array(np_data)
+                self._data = _variable(new_arr.data)
```

Comment:
I think `_variable` returns an `ivy.Array` even if we pass `new_arr.data` to it, so you'd have to convert it back to native before storing `self._data`.

Reply:
In that case something like `_variable(new_arr).data` should work right?